x-pack/plugins/alerting/server/alerts_client/alerts_client.test.ts

```diff
@@ -770,1217 +634,1371 @@ describe('Alerts Client', () => {
+                    uuid: 'abc',
                     workflow_status: 'open',
                   },
                   space_ids: ['default'],
-                  version: '8.8.0',
+                  version: '8.9.0',
                 },
                 tags: ['rule-', '-tags'],
               },
-            },
-          ],
-        },
-      });
-      const alertsClient = new AlertsClient<{}, {}, {}, 'default', 'recovered'>(alertsClientParams);
-
-      await alertsClient.initializeExecution({
-        maxAlerts,
-        ruleLabel: `test: rule-name`,
-        flappingSettings: DEFAULT_FLAPPING_SETTINGS,
-        activeAlertsFromState: {
-          '1': {
-            state: { foo: true, start: '2023-03-28T12:27:28.159Z', duration: '0' },
-            meta: {
-              flapping: false,
-              flappingHistory: [true],
-              maintenanceWindowIds: [],
-              lastScheduledActions: { group: 'default', date: new Date() },
-              uuid: 'abc',
-            },
-          },
-          '2': {
-            state: { foo: true, start: '2023-03-28T02:27:28.159Z', duration: '36000000000000' },
-            meta: {
-              flapping: false,
-              flappingHistory: [true, false],
-              maintenanceWindowIds: [],
-              lastScheduledActions: { group: 'default', date: new Date() },
-              uuid: 'def',
-            },
-          },
-        },
-        recoveredAlertsFromState: {},
-      });
-
-      // Report 1 new alert and 1 active alert, recover 1 alert
-      const alertExecutorService = alertsClient.factory();
-      alertExecutorService.create('2').scheduleActions('default');
-      alertExecutorService.create('3').scheduleActions('default');
-
-      alertsClient.processAndLogAlerts(processAndLogAlertsOpts);
-
-      await alertsClient.persistAlerts();
-
-      const { alertsToReturn } = alertsClient.getAlertsToSerialize();
-      const uuid3 = alertsToReturn['3'].meta?.uuid;
-
-      expect(clusterClient.bulk).toHaveBeenCalledWith({
-        index: '.alerts-test.alerts-default',
-        refresh: 'wait_for',
-        require_alias: true,
-        body: [
-          {
-            index: {
-              _id: 'def',
-              _index: '.internal.alerts-test.alerts-default-000002',
-              require_alias: false,
-            },
-          },
-          // ongoing alert doc
-          {
-            '@timestamp': date,
-            event: {
-              action: 'active',
-              kind: 'signal',
-            },
-            kibana: {
-              alert: {
-                action_group: 'default',
-                duration: {
-                  us: '72000000000000',
-                },
-                flapping: false,
-                flapping_history: [true, false, false],
-                instance: {
-                  id: '2',
+              {
+                create: { _id: uuid2, ...(useDataStreamForAlerts ? {} : { require_alias: true }) },
```

Comment:
One general change that was made was that for **new** alerts, we use `create` instead of `index`, since we can't use `index` to create new docs in data streams.  We continue to use `index` to **update** alert docs, but now need OCC info for that as well.